### PR TITLE
SIMPLE-5479 option for setting definition read only property

### PR DESCRIPTION
**Pull request by daniel-valent**
5377 needs to merge first

### CODE REVIEW

**Comment by tmikuska** [comment]
> 5377 needs to merge first

Can you separate this change, so we can merge it before 5377?  I assume that it will take time to review 5377 and it also may be too risky for 2.6.

**Comment by daniel-valent** [comment]
Hmm, I'll basically need to downgrade the url stuff, but yeah can do

**Comment by daniel-valent** [comment]
Done, ready for review

**Review by tmikuska — APPROVED** [review]
lgtm thanks

File: virl2_client/models/node_image_definitions.py [hunk 1]
```diff
@@ -82,6 +82,30 @@ def image_definitions_for_node_definition(self, definition_id: str) -> list[dict
...
+        self, definition_id: str, read_only: bool
+    ) -> None:
+        """
+        Set the read-only attribute of the image definition with the given ID.
+
+        :param definition_id: The ID of the image definition.
+        :param read_only: The new value of the read-only attribute.
+        """
+        url = f"image_definitions/{definition_id}/read_only"
+        self._session.put(url, json=read_only)
```
Comment (tmikuska) [inline virl2_client/models/node_image_definitions.py:95]: can we change these to return a response?  I just found out that both endpoints return a full def object.
```suggestion
        return self._session.put(url, json=read_only).json()
```

Comment (daniel-valent) [reply virl2_client/models/node_image_definitions.py]: Done.

**Pull request merged**